### PR TITLE
Migrated dependencies to version catalogs

### PR DESCRIPTION
**Pull request by Pelkar**
**Jira issue:**
https://glia.atlassian.net/browse/MOB-2460

**What was solved?**
- Migrated dependencies to version catalogs  

**Release notes:**

 - [ ] Feature
 - [X] Ignore
 - [ ] Release notes (Is it clear from the description here?)
 - [ ] Migration guide (If changes are needed for integrator already using the SDK - what needs to be communicated? Add underneath please)

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -85,14 +85,16 @@ apply from: "${rootDir}/scripts/version-updater.gradle"
...
-  appCompatVersion = '1.3.1'
-  materialVersion = '1.6.1'
+  appCompatVersion = '1.6.1'
+  materialVersion = '1.11.0'
   constraintLayoutVersion = '2.1.4'
   navVersion = '2.3.5'
-  lifecycle_process = '2.3.1'
-  rxAndroid = '2.1.0'
-  rxJava = '2.2.6'
+  lifecycle_process = '2.7.0'
```
Comment (andrews-moc) [inline build.gradle:97]: Shall the whole `ext` section be also migrated to a version catalog approach?

Comment (Pelkar) [reply build.gradle:97]: Yes, I will cleanup

File: app/build.gradle [hunk 2]
```diff
@@ -2,7 +2,7 @@ apply plugin: 'com.android.application'
 apply plugin: 'org.jetbrains.kotlin.android'
 
 android {
-    compileSdkVersion 33
+    compileSdkVersion 34
```
Comment (DavDo) [inline app/build.gradle:5]: As far as I'm informed, this property is deprecated and must be replaced to the `compileSdk 34`

File: widgetssdk/build.gradle [hunk 3]
```diff
@@ -4,7 +4,7 @@ apply plugin: 'kotlin-parcelize'
 apply plugin: 'app.cash.paparazzi'
 
 android {
-  compileSdkVersion 33
+  compileSdkVersion 34
```
Comment (DavDo) [inline widgetssdk/build.gradle:7]: We agreed to upgrade the compile SDK version to 33, so integrators aren't forced to upgrade to the latest version. Upgrading this to the latest available API version will force the integrators to upgrade their dependencies, especially `core-ktx` to the latest version. I'd prefer to keep the testing app compile and target SDK versions latest API version(34), and Widget and core sdk target and compile sdk versions one version lowe(33).

**Review by DavDo — CHANGES_REQUESTED** [review]
I think it's a great initiative. 
However, I suggest handling the migration to RxJava RX3 and the migration of dependencies to the version catalogs separately. Also, I don't recommend upgrading the compile version of the widgets SDK, as it may cause certain libraries to be updated to their latest versions.

File: widgetssdk/build.gradle (hunk 3 above)
Comment (andrews-moc) [reply widgetssdk/build.gradle:7]: Yes, indeed. I was about to write this also.

**Review by andrews-moc — APPROVED** [review]
Just update the PR name for clarity. 😉

**Pull request merged**